### PR TITLE
Add cache key prefix + ttl configuration

### PR DESCRIPTION
**Pull request by clifton**

### CODE REVIEW

File: lib/decorator.js [hunk 1]
```diff
@@ -18,8 +18,9 @@ module.exports = function cachedComponent(cacheKeyFn) {
 
     CachedComponent.cacheKeyFn = cacheKeyFn || JSON.stringify;
     CachedComponent.canCache = true;
+    CachedComponent.cacheTtlInSeconds = opts.cacheTtlInSeconds;
     CachedComponent.wrappedComponentName = wrappedComponentName;
-    CachedComponent.displayName = 'CachedComponent(' + wrappedComponentName + ')';
+    CachedComponent.displayName = `CachedComponent(${wrappedComponentName})`;
```
Comment (Peleg) [inline lib/decorator.js]: the only problem i had with this is that we're not transpiling this lib at all (most of it runs in node anyway) and if a client then requires this file, they will have to specifically make sure they compile it separately. not sure it's worth it just for this change tbh

File: lib/stores/MemcachedCacheStore.js [hunk 2]
```diff
@@ -8,13 +8,22 @@ module.exports = class MemcachedCacheStore extends CacheStore {
...
+    this.cacheKeyPrefix = opts.cacheKeyPrefix;
+    this.defaultTtlInSeconds = opts.defaultTtlInSeconds || 10 * 60;
+
     this._queuedKeys = [];
     this._scheduledMultiGet;
   }
 
+  _toCacheKey(key) {
+    return this.cacheKeyPrefix ?
+      `${this.cacheKeyPrefix}:${key}` :
```
Comment (Peleg) [inline lib/stores/MemcachedCacheStore.js]: this will be fine however since it runs on the server only

File: lib/stores/MemcachedCacheStore.js [hunk 3]
```diff
@@ -32,7 +41,7 @@ module.exports = class MemcachedCacheStore extends CacheStore {
 
   _multiGet(key) {
     return new Promise((res, rej) => {
-      this._queuedKeys.push(key);
+      this._queuedKeys.push(this._toCacheKey(key));
```
Comment (Peleg) [inline lib/stores/MemcachedCacheStore.js:44]: should we normalize the cache key here or just in the public `get` method of this class?

Comment (clifton) [reply lib/stores/MemcachedCacheStore.js:44]: Either way is good with me

File: package.json [hunk 4]
```diff
@@ -26,8 +26,13 @@
   },
   "homepage": "https://github.com/peleg/react-server-cache#readme",
   "devDependencies": {
-    "eslint": "^2.13.1",
-    "eslint-config-airbnb": "^9.0.1",
+    "babel-cli": "^6.23.0",
+    "babel-eslint": "^7.1.1",
```
Comment (Peleg) [inline package.json]: dont think we need those if not transpiling, no?

File: .eslintrc [hunk 5]
```diff
@@ -18,7 +18,8 @@
     "no-underscore-dangle": 0,
     "import/no-extraneous-dependencies": 0,
     "import/prefer-default-export": 0,
-    "no-unused-vars": 0
+    "no-unused-vars": 0,
+    "prefer-template": 0
```
Comment (Peleg) [inline .eslintrc:22]: 👍

File: lib/stores/MemcachedCacheStore.js [hunk 6]
```diff
@@ -8,13 +8,22 @@ module.exports = class MemcachedCacheStore extends CacheStore {
...
+    this.cacheKeyPrefix = opts.cacheKeyPrefix;
+    this.defaultTtlInSeconds = opts.defaultTtlInSeconds || 10 * 60;
+
     this._queuedKeys = [];
     this._scheduledMultiGet;
   }
 
+  _toCacheKey(key) {
+    return this.cacheKeyPrefix ?
+      [this.cacheKeyPrefix, key].join(':') :
```
Comment (Peleg) [inline lib/stores/MemcachedCacheStore.js:20]: you could use templates here still. just not on client side code. this is fine too

**Review by Peleg — APPROVED** [review]
👍

**Pull request merged**